py/htm/encoders/eye.py

```diff
@@ -0,0 +1,627 @@
+        output_diameter   = 200, # output SDR size is diameter^2
+        sparsityParvo     = 0.2,
+        sparsityMagno     = 0.025,
+        color             = True,):
+        """
+        Argument output_diameter is size of output ... output is a 
+            field of view (image) with circular shape. Default 200. 
+            `parvo/magno_sdr` size is `output_diameter^2`
+        Argument `sparsityParvo` - sparsity of parvo-cellular pathway of the eye.
+            As a simplification, "parvo" cells (P-cells) represent colors, static 
+            object's properties (shape,...) and are used for image classification. 
+            For biologically accurate details see eg. 
+            https://foundationsofvision.stanford.edu/chapter-5-the-retinal-representation/
+            Note: biologically, the ratio between P/M-cells is about 8(P):1(M):(1 rest), see
+            https://www.pnas.org/content/94/11/5900
+        Argument `sparsityMagno` - sparsity of the magno-cellular (M-cells) pathway, which
+            transfers (higly) temporal information in the visual data, as use "used" for
+            motion detection and motion tracking, video processing.
+            For details see @param `sparsityParvo`.
+            TODO: output of M-cells should be processed on a fast TM.
+        Argument color: use color vision (requires P-cells > 0), default true.
+        """
+        self.output_diameter   = output_diameter
+        # Argument resolution_factor is used to expand the sensor array so that
+        # the fovea has adequate resolution.  After log-polar transform image
+        # is reduced by this factor back to the output_diameter.
+        self.resolution_factor = 3
+        self.retina_diameter   = int(self.resolution_factor * output_diameter)
+        # Argument fovea_scale  ... proportion of the image (ROI) which will be covered (seen) by
+        # high-res fovea (parvo pathway)
+        self.fovea_scale       = 0.177
+        assert(output_diameter // 2 * 2 == output_diameter) # Diameter must be an even number.
+        assert(self.retina_diameter // 2 * 2 == self.retina_diameter) # (Resolution Factor X Diameter) must be an even number.
+        assert(sparsityParvo >= 0 and sparsityParvo <= 1.0)
+        if sparsityParvo > 0:
+          assert(sparsityParvo * (self.retina_diameter **2) > 0)
+        self.sparsityParvo = sparsityParvo
+        assert(sparsityMagno >= 0 and sparsityMagno <= 1.0)
+        if sparsityMagno > 0:
+          assert(sparsityMagno * (self.retina_diameter **2) > 0)
+        self.sparsityMagno = sparsityMagno
+        if color is True:
+          assert(sparsityParvo > 0)
+        self.color = color
+
+
+        self.retina = cv2.bioinspired.Retina_create(
+            inputSize            = (self.retina_diameter, self.retina_diameter),
+            colorMode            = color,
+            colorSamplingMethod  = cv2.bioinspired.RETINA_COLOR_BAYER,
+            useRetinaLogSampling = True,)
```

Comment:
@ctrl-z-9000-times please compare with this on/off. Can it replace our manual log-polar transformation?